Hakawai/Mentions/HKWMentionsStartDetectionStateMachine.m

```diff
@@ -194,18 +204,20 @@ - (void)characterTyped:(unichar)c asInsertedCharacter:(BOOL)inserted previousCha
 }
 
 - (void)deleteTypedCharacter:(unichar)deletedChar withCharacterNowPrecedingCursor:(unichar)precedingChar {
-    NSCharacterSet *whitespaceSet = [NSCharacterSet whitespaceAndNewlineCharacterSet];
     // Determine the type of the character
     enum CharacterType {
-        CharacterTypeWhitespace = 0,
+        CharacterTypeSeparator = 0,
         CharacterTypeNormal
     };
-    enum CharacterType deletedCharacterType = ([whitespaceSet characterIsMember:deletedChar] || deletedChar == 0)
-                                                ? CharacterTypeWhitespace
-                                                : CharacterTypeNormal;
-    enum CharacterType currentCharacterType = ([whitespaceSet characterIsMember:precedingChar] || precedingChar == 0)
-                                                ? CharacterTypeWhitespace
-                                                : CharacterTypeNormal;
+    enum CharacterType deletedCharacterType = ([HKWMentionsStartDetectionStateMachine.whitespaceSet characterIsMember:deletedChar] || deletedChar == 0)
```

Comment:
why do these two have different logics now? before they were the same

Reply:
Made changes to set `deletedCharacterType` with respect to separator set just like for `currentCharacterType`.